packages/node/src/soroban/block.soroban.ts

```diff
@@ -57,24 +57,43 @@ export class SorobanBlockWrapped implements SorobanBlockWrapper {
     return true;
   }
 
-  static decodeScVals(value: xdr.ScVal): any {
-    switch (value.switch()) {
-      case xdr.ScValType.scvBytes(): {
-        const buffer = value.bytes();
-        return buffer.toString();
-      }
-      case xdr.ScValType.scvSymbol(): {
-        return value.sym().toString();
-      }
+  static decodeScVals(scVal: xdr.ScVal): any {
```

Comment:
Does soroban not provide this functionality?

Reply:
their `scValToNative` had issues for some datatypes, such as ScvString.

Reply:
Is it worth contributing back to soroban sdk?

Reply:
The issue with ScvString was we get a Uint8Array. From the docs, it looks it was intended to be that way: 

```string -> string IF the underlying buffer can be decoded as ascii/utf8, Uint8Array of the raw contents in any error case```

so, we just extract the Uint8Array and decode it from base64 to utf8